sqlmesh/core/metric/definition.py

```diff
@@ -0,0 +1,193 @@
+from __future__ import annotations
+
+import typing as t
+from pathlib import Path
+
+from pydantic import validator
+from sqlglot import exp
+from sqlglot.helper import first
+
+from sqlmesh.core import dialect as d
+from sqlmesh.utils import UniqueKeyDict
+from sqlmesh.utils.errors import ConfigError
+from sqlmesh.utils.pydantic import PydanticModel
+
+
+def load_metric_ddl(
+    expression: exp.Expression, dialect: t.Optional[str], path: Path = Path(), **kwargs: t.Any
+) -> MetricMeta:
+    """Returns a MetricMeta from raw Metric DDL."""
+    if not isinstance(expression, d.Metric):
+        _raise_metric_config_error(
+            f"Only METRIC(...) statements are allowed. Found {expression.sql(pretty=True)}", path
+        )
+
+    metric = MetricMeta(
+        **{
+            "dialect": dialect,
+            "description": "\n".join(comment.strip() for comment in expression.comments)
+            if expression.comments
+            else None,
+            **{prop.name.lower(): prop.args.get("value") for prop in expression.expressions},
+            **kwargs,
+        }
+    )
+
+    metric._path = path
+
+    return metric
+
+
+def expand_metrics(metas: UniqueKeyDict[str, MetricMeta]) -> UniqueKeyDict[str, Metric]:
+    """Resolves all metas into standalone metrics."""
+    metrics: UniqueKeyDict[str, Metric] = UniqueKeyDict("metrics")
+
+    for name, meta in metas.items():
+        if name not in metrics:
+            metrics[name] = meta.to_metric(metas, metrics)
+
+    return metrics
+
+
+def find_tables(expression: exp.Expression, dialect: str) -> t.Set[str]:
```

Comment:
[Nit] Should this be prefixed with `_`? We have a function with the exact same name already and it might not be a bad idea to indicate that this one is specific to just this module.

Reply:
will do